The data is quite messy. In particular, institution names have not been entered consistently by all users leading to various nodes in the consturcted graph actually referring to the same institution. This is an attempt to clean this dataset. 

Read in the data

In [22]:
import pandas as pd
tdf_all = pd.read_csv('../data/edgelists/tdf_all.csv')
tdf_all = tdf_all.dropna()
tdf_all

node_u  \
0                             National Capital Consortium   
1                             National Capital Consortium   
2                                  Albany Medical College   
3       University of Texas Health Science Center San ...   
4       University of Texas Health Science Center San ...   
...                                                   ...   
502802    McGaw Medical Center of Northwestern University   
502803  Rush Medical College of Rush University Medica...   
502804                         Westchester Medical Center   
502805                         Westchester Medical Center   
502806                        Stanley Medical College SMC   

                                                   node_v  timestamp  \
0                             National Capital Consortium     1972.0   
1                             National Capital Consortium     1968.0   
2                             National Capital Consortium     1967.0   
3                                University of Washington     2003.0   
4       University of Texas Health Science Center San ...     1999.0   
...                                                   ...        ...   
502802    McGaw Medical Center of Northwestern University     2000.0   
502803    McGaw Medical Center of Northwestern University     1997.0   
502804                                Stanford University     2018.0   
502805                         Westchester Medical Center     2015.0   
502806                         Westchester Medical Center     2011.0   

        doctor_id     specialty  weight  
0          3721.0  neurosurgery       1  
1          3721.0  neurosurgery       1  
2          3721.0  neurosurgery       1  
3          2833.0  neurosurgery       1  
4          2833.0  neurosurgery       1  
...           ...           ...     ...  
502802      182.0    hematology       1  
502803      182.0    hematology       1  
502804      478.0    hematology       1  
502805      478.0    hematology       1  
502806      478.0    hematology       1  

[502806 rows x 6 columns]

Yay it worked. Now lets look at all the unique institutions

In [23]:
unique_institutions = pd.concat([tdf_all['node_v'], tdf_all['node_u']]).unique()
len(unique_institutions)

11738

11738 unique institutions! Thats a lot! Lets see what they are

In [ ]:
unique_institutions = list(unique_institutions)


In [26]:
import numpy as np

sorted_unique_institutions = np.sort(unique_institutions)
list(sorted_unique_institutions)

['\tUniversidad Tecnólogica Equinoccial Facultad de Ciencias de la Salud Eugenio Espejo',
 ' American University of Beirut',
 ' Autonomous University of Sinaloa ',
 ' Belarusian State Medical University ',
 ' Creighton University',
 ' Harvard T.H. Chan School of Public Health',
 ' Institute of Medical Sciences, Banaras Hindu University, Varanasi, INDIA ',
 ' Johns Hopkins University School of Medicine',
 ' Letterman Army Medical Center ',
 ' Los Angeles Psychoanalytic Institute ',
 ' M.D. Anderson Cancer Center',
 ' Massachusetts General Hospital',
 ' Medical Scientist Training Program at Icahn School of Medicine at Mount Sinai ',
 ' Northwestern University',
 ' Notre Dame ',
 ' Ross University School of medicine ',
 " Schneider Children's Hospital / Long Island Jewish Medical Center",
 ' Stanford University School of Medicine',
 ' Tufts College',
 ' University of Cincinnati Medical Center/College of Medicine University of Cincinnati Medical Center/College of Medicine',
 ' University o

In [ ]:
from rapidfuzz import process, fuzz

institutions = unique_institutions

canonical_names = [
    "Brigham and Women's Hospital",
]

# Define a similarity threshold
threshold = 85  # Adjust based on how strict you want the match to be

# Map to store results
grouped_institutions = {}

# Match each string to the most similar canonical name
for institution in institutions:
    match, score, *_ = process.extractOne(institution, canonical_names, scorer=fuzz.ratio)
    if score >= threshold:
        grouped_institutions[institution] = match
    else:
        # If no match is above the threshold, treat it as a new canonical name
        canonical_names.append(institution)
        grouped_institutions[institution] = institution

grouped_institutions

{'National Capital Consortium': 'National Capital Consortium',
 'University of Washington': 'University of Washington',
 'University of Texas Health Science Center San Antonio Joe and Teresa Lozano Long School of Medicine': 'University of Texas Health Science Center San Antonio Joe and Teresa Lozano Long School of Medicine',
 'University of California (San Francisco) School of Medicine': 'University of California (San Francisco) School of Medicine',
 'University of Oklahoma Health Sciences Center': 'University of Oklahoma Health Sciences Center',
 'Indiana University School of Medicine': 'Indiana University School of Medicine',
 'University of Miami/Jackson Health System': 'University of Miami/Jackson Health System',
 'UCLA David Geffen School of Medicine/UCLA Medical Center': 'UCLA David Geffen School of Medicine/UCLA Medical Center',
 'Vanderbilt University Medical Center': 'Vanderbilt University Medical Center',
 'Lenox Hill Hospital-Northwell Health': 'Lenox Hill Hospital-Northwell

In [35]:
len(canonical_names)

7998

In [42]:
b_inst = [institution for institution in unique_institutions if institution.lower().startswith('b')]
len(b_inst)

463

In [ ]:
canonical_names = [
    
]

In [55]:
fuzz.ratio("BAYLOR", "baylor")
fuzz.token_ratio("BAYLOR", "BAYlro")

50.0

In [109]:
from collections import Counter
institutions = list(pd.concat([tdf_all['node_v'], tdf_all['node_u']]))
institution_counts = dict(Counter(institutions))
institution_counts
sorted_institution_counts = dict(sorted(institution_counts.items(), key=lambda item: item[1], reverse=True))
b_institution_counts = {k: v for k, v in sorted_institution_counts.items() if k.lower().startswith('b')}

In [70]:
sorted_institution_counts

{'Baylor College of Medicine': 9050,
 'Emory University School of Medicine': 8837,
 'Johns Hopkins University': 7560,
 'NYU Grossman School of Medicine': 7507,
 'Yale-New Haven Medical Center': 7237,
 'Washington University/B-JH/SLCH Consortium': 7144,
 'Indiana University School of Medicine': 7058,
 'SUNY Downstate Health Sciences University': 6939,
 'University of Texas Southwestern Medical Center': 6885,
 'UPMC Medical Education': 6539,
 'Mayo Clinic College of Medicine and Science (Rochester)': 6532,
 'University of California (San Francisco)': 6428,
 'Duke University Hospital': 6424,
 'UCLA David Geffen School of Medicine/UCLA Medical Center': 6071,
 'New York Presbyterian Hospital (Cornell Campus)': 5976,
 'Icahn School of Medicine at Mount Sinai/Mount Sinai Hospital': 5945,
 'McGaw Medical Center of Northwestern University': 5805,
 'University of Michigan': 5740,
 'University of Miami/Jackson Health System': 5594,
 'University of Pennsylvania Health System': 5353,
 'Cleveland Cl

In [114]:
b_institution_counts

{'Baylor College of Medicine': 9050,
 'Boston University Medical Center': 4483,
 'Beth Israel Deaconess Medical Center': 4063,
 'Brown University': 3338,
 "Brigham and Women's Hospital": 3204,
 'Boston University School of Medicine': 2061,
 'BronxCare Health System': 1756,
 "Brigham and Women's Hospital/Harvard Medical School": 1547,
 'Baylor University Medical Center': 1243,
 'Brooklyn Hospital Center': 1220,
 'Boston Children’s Hospital/Boston Medical Center': 1024,
 "Brigham and Women's Hospital/Massachusetts General Hospital/Harvard Medical School": 771,
 'Baylor College of Medicine (Houston)': 699,
 'Brody School of Medicine at East Carolina University': 615,
 'Butler Hospital/Brown University': 558,
 'Bridgeport Hospital/Yale University': 530,
 "Barrow Neurological Institute at St Joseph's Hospital and Medical Center": 449,
 'Bassett Medical Center': 433,
 'Boston VA Healthcare System (Brockton-West Roxbury)': 418,
 'Boston Children’s Hospital': 417,
 'Baroda Medical College': 41

In [116]:
b_inst = list(b_institution_counts.keys())
len(b_inst)

463

In [177]:
# with open('../data/b_inst.txt', 'w') as file:
#     for item in b_inst:
#         file.write(f"{item}\n")

b_inst = []
with open('../data/b_inst.txt', 'r') as file:
    for line in file:
        name = line.replace('\n', '')
        b_inst.append(name)

In [76]:

b_inst
cleaned_b_inst = []
for inst in b_inst:
    inst = inst.strip()  # Remove leading and trailing whitespaces
    inst = inst.replace('-', ' ')  # Replace '-' with a space
    if '/' in inst:
        inst = inst.split('/')[0]  # Take only the part before the slash
    # Remove specific words
    # for word in ['hospital', 'university', 'medical', 'college', 'school']:
    #     inst = inst.lower().replace(word, '')
    cleaned_b_inst.append(inst)
b_inst = list(set(cleaned_b_inst))


In [77]:
len(b_inst)

406

In [216]:
from rapidfuzz import process, fuzz

institutions = b_inst

canonical_names = [
    "Brigham and Women's Hospital",
]

# Define a similarity threshold
threshold = 60  # Adjust based on how strict you want the match to be

# Map to store results
grouped_institutions = {}

# Match each string to the most similar canonical name
for institution in institutions:
    match, score, *_ = process.extractOne(institution, canonical_names, scorer=fuzz.token_ratio)
    if score >= threshold:
        grouped_institutions[institution] = match
    else:
        # If no match is above the threshold, treat it as a new canonical name
        canonical_names.append(institution)
        grouped_institutions[institution] = institution

len(grouped_institutions)

463

In [217]:
grouped_institutions

{'B I Deaconess Med Ctr': 'B I Deaconess Med Ctr',
 'B J Medical College Gujarat University.': 'B J Medical College Gujarat University.',
 'B J Medical college, Ahmedabad , India': 'B J Medical college, Ahmedabad , India',
 'B Palmer Eye Institute - A B Leach Eye': 'B Palmer Eye Institute - A B Leach Eye',
 'B. J. Medical College Ahmedabad': 'B J Medical college, Ahmedabad , India',
 'B.J. Medical College - University of Pune': 'B J Medical College Gujarat University.',
 "B.J. Wadia Children's Hospital": "Brigham and Women's Hospital",
 'B.P. Koirala Institute of Health Sciences': 'B.P. Koirala Institute of Health Sciences',
 'BAMC': 'BAMC',
 'BAYLOR': 'BAYLOR',
 'BENEDICTINE UNIV': 'BENEDICTINE UNIV',
 'BLDE University - Medical College': 'B J Medical College Gujarat University.',
 'BMU Karachi': 'BMU Karachi',
 'BRIGHAM YOUNG UNIVERSITY': 'BRIGHAM YOUNG UNIVERSITY',
 'BROWN UNIVERSITY, School for Professional Studies': 'BROWN UNIVERSITY, School for Professional Studies',
 'BYU': 'BYU

In [179]:
my_dict = {
  "B I Deaconess Med Ctr": [
    "B I Deaconess Med Ctr",
    "Beth Israel Deaconess Medical Center",
    "Beth Israel Deaconess Medical Center (Harvard)",
    "Beth Israel Deaconess Medical Center - Harvard Medical School",
    "Beth Israel Deaconess Medical Center, Harvard Medical School",
    "Beth Israel Deaconess Medical Center/Children's Hospital/Harvard Medical School",
    "Beth Israel Deaconess Medical Center/Harvard Medical School"
  ],
  "B J Medical College": [
    "B J Medical College Gujarat University.",
    "B J Medical college, Ahmedabad , India",
    "B. J. Medical College Ahmedabad",
    "B.J. Medical College - University of Pune"
  ],
  "B Palmer Eye Institute": [
    "B Palmer Eye Institute - A B Leach Eye",
    "Bascom Palmer Eye Institute",
    "Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami",
    "Bascom Palmer Eye Institute, University of Miami"
  ],
  "B.J. Wadia Children's Hospital": [
    "B.J. Wadia Children's Hospital"
  ],
  "B.P. Koirala Institute of Health Sciences": [
    "B.P. Koirala Institute of Health Sciences"
  ],
  "BAMC": [
    "BAMC"
  ],
  "BAYLOR": [
    "BAYLOR",
    "Baylor Affiliated Hospitals",
    "Baylor All Saints Medical Center Fort Worth",
    "Baylor Coll Med-Houston",
    "Baylor College of Medicine",
    "Baylor College of Medicine (Houston)",
    "Baylor College of Medicine (San Antonio)",
    "Baylor College of Medicine / Texas Children's Hospital",
    "Baylor College of Medicine/Cullen Eye Institute",
    "Baylor College of Medicine/St Luke's Episcopal Hospital",
    "Baylor College of Medicine/Texas Children's Hospital",
    "Baylor College of Medicine/Texas Heart Institute",
    "Baylor College of Medicine/Texas Heart Institute/Baylor St Luke's Medical Center",
    "Baylor College of medicine",
    "Baylor High School",
    "Baylor Institute of Immunology Research",
    "Baylor Medical Center",
    "Baylor Medical Center at Garland",
    "Baylor Scott & White Medical Center - Temple",
    "Baylor Scott and White Medical Center - Round Rock",
    "Baylor Scott and White The Heart Hospital Plano",
    "Baylor U Med Ctr",
    "Baylor University",
    "Baylor University Medical Center",
    "Baylor University Medical Center/Texas Heart Hospital Baylor Plano",
    "Baylor college of medicine",
    "Baylor university",
    "Baylor-Texas Heart"
  ],
  "BENEDICTINE UNIV": [
    "BENEDICTINE UNIV",
    "Benedictine College",
    "Benedictine Hospital",
    "Benedictine University"
  ],
  "BLDE University - Medical College": [
    "BLDE University - Medical College"
  ],
  "BMU Karachi": [
    "BMU Karachi"
  ],
  "BRIGHAM YOUNG UNIVERSITY": [
    "BRIGHAM YOUNG UNIVERSITY",
    "BYU",
    "Brigham Young Univerdity",
    "Brigham Young University",
    "Brigham Young University - Idaho",
    "Brigham Young University-Hawaii",
    "Brigham young university"
  ],
  "BROWN UNIVERSITY": [
    "BROWN UNIVERSITY, School for Professional Studies",
    "Brown",
    "Brown University",
    "Brown Univeristy",
    "Brown Univerity",
    "Brown University (Butler Hospital)",
    "Brown University Medical School",
    "Brown University School of Public Health",
    "Brown University, Associate Professor of Medicine",
    "Brown University, Class of 1982",
    "Brown University/ Women and Infants Hospital",
    "Brown University/Rhode Island Hospital-Lifespan",
    "Brown University/Women & Infants Hospital",
    "Brown/Rhode Island Hospital",
    "brown university"
  ],
  "Baba Raghav Das Medical College": [
    "Baba Raghav Das Medical College"
  ],
  "Babson College": [
    "Babson College",
    "Babson College Olin Business School",
    "Babson/Olin School of Management"
  ],
  "Baghdad University Medical College": [
    "Baghdad University Medical College"
  ],
  "Bahcesehir University": [
    "Bahcesehir University"
  ],
  "Baker College": [
    "Baker College"
  ],
  "Baldwin Wallace University": [
    "Baldwin Wallace University",
    "Baldwin Wallace University",
    "Baldwin-Wallace College"
  ],
  "Ball State University": [
    "Ball State University"
  ],
  "Balliol College": [
    "Balliol College"
  ],
  "Baltimore City Community College": [
    "Baltimore City Community College"
  ],
  "Baltimore City Hospital": [
    "Baltimore City Hospital"
  ],
  "Baltimore County Fire-Rescue Academy": [
    "Baltimore County Fire-Rescue Academy"
  ],
  "Baltimore Medical College": [
    "Baltimore Medical College"
  ],
  "Baltimore University": [
    "Baltimore University"
  ],
  "Baltimore Washington Institute for Psychoanalysis": [
    "Baltimore Washington Institute for Psycho Analysis",
    "Baltimore Washington Institute for Psychoanalysis",
    "Baltimore-Washington Institute for Psychoanalysis",
    "Baltimore-Washington Psychoanalytic Institute"
  ],
  "Banaras Hindu University IOM": [
    "Banaras Hindu University IOM"
  ],
  "Banff School of Management": [
    "Banff School of Management"
  ],
  "Bangabandhu Shiekh Mujib Medical University": [
    "Bangabandhu Shiekh Mujib Medical University"
  ],
  "Bangalore Medical College": [
    "Bangalore Medical College",
    "Bangalore Medical School",
    "Bangalore Medical college",
    "Bangalore Medical college and Research Institute"
  ],
  "Bangalore University": [
    "Bangalore University"
  ],
  "Bangladesh College of Physicians and Surgeons": [
    "Bangladesh College of Physicians and Surgeons"
  ],
  "Bangladesh Medical College": [
    "Bangladesh Medical College"
  ],
  "Bank Street College of Education": [
    "Bank Street College of Education"
  ],
  "Bankstown-Lidcombe Hospital": [
    "Bankstown-Lidcombe Hospital"
  ],
  "Banner University Medical Center": [
    "Banner - University Medical Center Phoenix",
    "Banner - University of Arizona",
    "Banner University Medical Center",
    "Banner University Medical Center Phoenix",
    "Banner University of Arizona",
    "Banner-University Medical Center Phoenix"
  ],
  "Baptist Health": [
    "Baptist Health",
    "Baptist Health - Deaconess Madisonville",
    "Baptist Health - UAMS",
    "Baptist Health System",
    "Baptist Health Systems"
  ],
  "Baptist Hospital": [
    "Baptist Hospital",
    "Baptist Hospital of Miami",
    "Baptist Hospital of Pensacola"
  ],
  "Baptist Medical Center": [
    "Baptist Medical Center",
    "Baptist Medical Center",
    "Baptist Medical Center Jacksonville",
    "Baptist Medical Center of New York"
  ],
  "Baptist Mem Hosp": [
    "Baptist Mem Hosp"
  ],
  "Baptist Memorial Medical Education": [
    "Baptist Memorial Medical Education",
    "Baptist Memorial Medical Education (Memphis)"
  ],
  "Baptist Neurological Institute": [
    "Baptist Neurological Institute"
  ],
  "Baptist Outreach Services (Montgomery)": [
    "Baptist Outreach Services (Montgomery)"
  ],
  "Baqai Medical and Dental College": [
    "Baqai Medical and Dental College"
  ],
  "Bar Ilan University": [
    "Bar Ilan University"
  ],
  "Bard College": [
    "Bard College"
  ],
  "Barnard College": [
    "Barnard College",
    "Barnard College Columbia University",
    "Barnard College of Columbia University",
    "Barnard College of Columbia University",
    "Barnard College, Columbia U.",
    "Barnard College, Columbia University",
    "Barnard College, Columbia University"
  ],
  "Barnert Hospital": [
    "Barnert Hospital"
  ],
  "Barnes-Jewish Hospital": [
    "Barnes Hospital",
    "Barnes Jewish Hospital",
    "Barnes-Jewish Hospital"
  ],
  "Baroda Medical College": [
    "Baroda Medical College",
    "Baroda Medical College and S.S.G. Hospital",
    "Baroda medical college"
  ],
  "Barraquer Clinic": [
    "Barraquer Clinic"
  ],
  "Barraquer Institute of America": [
    "Barraquer Institute of America"
  ],
  "Barrett, the Honors College at Arizona State University": [
    "Barrett, the Honors College at Arizona State University"
  ],
  "Barrow Neurological Institute": [
    "Barrow Neurological Institute",
    "Barrow Neurological Institute",
    "Barrow Neurological Institute at Phoenix Children's",
    "Barrow Neurological Institute at St Joseph's Hospital and Medical Center",
    "Barrow Neurological Institute at St Josephs Hospital and Medical Center"
  ],
  "Barry University": [
    "Barry University",
    "Barry University",
    "Barry university"
  ],
  "Barts and The London SOM & Dent": [
    "Barts and The London SOM & Dent"
  ],
  "Baruch College": [
    "Baruch College Zicklin School of Business",
    "Baruch College/ CUNY",
    "Baruch college/ Mt Sinai School of Medicine"
  ],
  "Barzilai Medical Center": [
    "Barzilai Medical Center"
  ],
  "Bashkir State Med Univ": [
    "Bashkir State Med Univ"
  ],
  "Basildon and Thurrock General Hospital": [
    "Basildon and Thurrock General Hospital"
  ],
  "Bassett HealthCare": [
    "Bassett HealthCare",
    "Bassett Medical Center"
  ],
  "Bassin Center for Plastic Surgery": [
    "Bassin Center for Plastic Surgery"
  ],
  "Bates College": [
    "Bates College"
  ],
  "Baton Rouge General Medical Center": [
    "Baton Rouge General",
    "Baton Rouge General Medical Center"
  ],
  "BayCare Health System": [
    "BayCare Health System"
  ],
  "Bayamón Medical Center": [
    "Bayamón Medical Center"
  ],
  "Bayfront Health St. Petersburg": [
    "Bayfront Health St Petersburg",
    "Bayfront Health St. Petersburg"
  ],
  "Bayrische Julius-Maximilans Universitaet Wuerzburg": [
    "Bayrische Julius-Maximilans Universitaet Wuerzburg"
  ],
  "Baystate Medical Center": [
    "Baystate Medical Center",
    "Baystate Medical Center, Springfield, MA"
  ],
  "Bearwood College": [
    "Bearwood College"
  ],
  "Beasley School of Law Temple University": [
    "Beasley School of Law Temple University"
  ],
  "Beaumont Health": [
    "Beaumont Health",
    "Beaumont Health (Dearborn & Taylor)",
    "Beaumont Health (Dearborn)",
    "Beaumont Health (Farmington Hills)",
    "Beaumont Health (Royal Oak & Dearborn)",
    "Beaumont Health (Royal Oak)",
    "Beaumont Hospital",
    "Beaumont Hospital, Dearborn",
    "Beaumont Hospital, Dublin, Ireland",
    "Beaumont Hospital, Grosse Pointe",
    "Beaumont Hospital, Royal Oak",
    "Beaumont Royal Oak"
  ],
  "Bedford Veterans Affairs Medical Center, Edith Nourse Rogers Memorial Veterans Hospital": [
    "Bedford Veterans Affairs Medical Center, Edith Nourse Rogers Memorial Veterans Hospital"
  ],
  "Beer Yaakov - Ness Ziona Mental Health Center, Tel Aviv University": [
    "Beer Yaakov - Ness Ziona Mental Health Center, Tel Aviv University"
  ],
  "Beeson Aesthetic Surgery Institute": [
    "Beeson Aesthetic Surgery Institute"
  ],
  "Behavioral Tech": [
    "Behavioral Tech"
  ],
  "Beijing College of Acupuncture and Orthopedics": [
    "Beijing College of Acupuncture and Orthopedics"
  ],
  "Beijing Medical University": [
    "Beijing Medical University"
  ],
  "Beijing Tiantan Hospital": [
    "Beijing Tiantan Hospital"
  ],
  "Beilinson Hospital": [
    "Beilinson Hospital"
  ],
  "Beirut Arab University Faculty of Medicine": [
    "Beirut Arab University Faculty of Medicine"
  ],
  "Belarusian State Medical University": [
    "Belarusian State Medical University"
  ],
  "Bella Vista Hospital": [
    "Bella Vista Hospital"
  ],
  "Bellarmine University": [
    "Bellarmine College",
    "Bellarmine University"
  ],
  "Bellevue Hospital": [
    "Bellevue - N.Y.U. Medical Center",
    "Bellevue Hospital",
    "Bellevue Hospital (Columbia Service) and Manhattan V.A",
    "Bellevue Hospital Center - NYU Medical Center",
    "Bellevue Hospital Center NY",
    "Bellevue Hospital, NYU 3rd & 4th Division Medicine",
    "Bellevue Hospital-New York University"
  ],
  "Bellevue College": [
    "Bellevue College"
  ],
  "Bellevue University": [
    "Bellevue University"
  ],
  "Belmont Center for Comprehensive Treatment": [
    "Belmont Center for Comprehensive Treatment"
  ],
  "Belmont Psychiatric center": [
    "Belmont Psychiatric center"
  ],
  "Belvedere College": [
    "Belvedere College"
  ],
  "Ben Gurion University": [
    "Ben Gurion University",
    "Ben Gurion University",
    "Ben Gurion University Israel",
    "Ben Gurion University of the Negev",
    "Ben-Gurion University of the Negev"
  ],
  "Benha University": [
    "Benha Faculty of Medicine",
    "Benha University"
  ],
  "Benilde High School": [
    "Benilde High School"
  ],
  "Benjamin Franklin High School": [
    "Benjamin Franklin High School"
  ],
  "Bennett College": [
    "Bennett College"
  ],
  "Bennett Surgery Center": [
    "Bennett Surgery Center"
  ],
  "Benning Martin Army Community Hospital": [
    "Benning Martin Army Community Hospital"
  ],
  "Bennington College": [
    "Bennington College"
  ],
  "Beraja Medical Institute": [
    "Beraja Medical Institute"
  ],
  "Berenson Allen Center for Neurostimulation": [
    "Berenson Allen Center for Neurostimulation"
  ],
  "Bergen County Academies": [
    "Bergen County Academies"
  ],
  "Bergen New Bridge Medical Center": [
    "Bergen New Bridge Medical Center"
  ],
  "Berklee College of Music": [
    "Berklee College of Music"
  ],
  "Berkshire Medical Center": [
    "Berkshire Medical Center"
  ],
  "Berkshire Psychoanalytic Institute": [
    "Berkshire Psychoanalytic Institute"
  ],
  "Berry College": [
    "Berry College"
  ],
  "Beth Israel Hospital": [
    "Beth Israel Hospital"
  ],
  "Beth Israel Lahey Health": [
    "Beth Israel Lahey Health"
  ],
  "Beth Israel Medical Center": [
    "Beth Israel Medical Center",
    "Beth Israel Medical Center-North Division",
    "Beth Israel Medical Center-North Division. Albert Einstein College of Medicine."
  ],
  "Bethany College - Kansas": [
    "Bethany College - Kansas"
  ],
  "Bethany Medical Center": [
    "Bethany Medical Center"
  ],
  "Bethel College - Kansas": [
    "Bethel College - Kansas"
  ],
  "Bethel University": [
    "Bethel University"
  ],
  "Bethesda Eye Institute St. Louis University": [
    "Bethesda Eye Institute St. Louis University"
  ],
  "Bethesda Hospital": [
    "Bethesda Hospital"
  ],
  "Bethesda Naval Hospital": [
    "Bethesda Naval Hospital"
  ],
  "Beverly Hills-private practice": [
    "Beverly Hills-private practice"
  ],
  "Beverly Hospital": [
    "Beverly Hospital"
  ],
  "Bexar County Teaching Hospitals": [
    "Bexar County Teaching Hospitals"
  ],
  "Beyoglu Eye Training and Research Hospital": [
    "Beyoglu Eye Training and Research Hospital"
  ],
  "Bharti Vidyapeeth Medical College": [
    "Bharti Vidyapeeth Medical College"
  ],
  "Bhopal University": [
    "Bhopal University"
  ],
  "Bialystok Medical Academy": [
    "Bialystok Medical Academy"
  ],
  "Big Bend Community College": [
    "Big Bend Community College"
  ],
  "Bildungszentrum St. Konrad": [
    "Bildungszentrum St. Konrad"
  ],
  "Billings Clinic": [
    "Billings Clinic"
  ],
  "Binghamton University": [
    "Binghamton State University",
    "Binghamton University",
    "Binghamton University",
    "Binghamton University - SUNY",
    "Binghamton University, State University of New York"
  ],
  "Binzhou Medical University": [
    "Binzhou Medical University"
  ],
  "Biochemistry and Mathematics, UCLA": [
    "Biochemistry and Mathematics, UCLA"
  ],
  "Biola University": [
    "Biola University"
  ],
  "Birla Institute of Technology": [
    "Birla Institute of Technology"
  ],
  "Birmingham Psychiatry Rotation": [
    "Birmingham Psychiatry Rotation"
  ],
  "Birmingham School of Law": [
    "Birmingham School of Law"
  ],
  "Birmingham University Medical School": [
    "Birmingham University Medical School"
  ],
  "Birmingham-Southern College": [
    "Birmingham-Southern College"
  ],
  "Bishop Eustace Preparatory School": [
    "Bishop Eustace Preparatory School"
  ],
  "Black Hills State University": [
    "Black Hills State University"
  ],
  "Blake Medical Center": [
    "Blake Medical Center"
  ],
  "Bloomberg School of Public Health": [
    "Bloomberg School of Public Health",
    "Bloomberg School of Public Health The Johns Hopkins University",
    "Bloomberg School of Public Health, The Johns Hopkins University"
  ],
  "Bloomsburg University of Pennsylvania": [
    "Bloomsburg University of Pennsylvania"
  ],
  "Blythedale Children's Hospital": [
    "Blythedale Children's Hospital"
  ],
  "Bnai Zion Hospital": [
    "Bnai Zion Hospital"
  ],
  "Board of Intermediate Education-Karachi": [
    "Board of Intermediate Education-Karachi"
  ],
  "Board-certified dermatologist": [
    "Board-certified dermatologist"
  ],
  "Bob Jones University": [
    "Bob Jones University",
    "Bob jones university"
  ],
  "Bocconi University": [
    "Bocconi University"
  ],
  "Body Sculpting Cosmetic Surgery Center": [
    "Body Sculpting Cosmetic Surgery Center"
  ],
  "Boise State University": [
    "Boise State University"
  ],
  "Bolan Medical College": [
    "Bolan Medical College"
  ],
  "Bombay university": [
    "Bombay university"
  ],
  "Bon Secours Hospital": [
    "Bon Secours Hospital",
    "Bon Secours Mary Immaculate Hospital"
  ],
  "Booth Memorial Medical Center": [
    "Booth Memorial Medical Center",
    "booth memorial"
  ],
  "Borland Groover Clinic, Baptist Medical Center, St. Vincent Medical Center": [
    "Borland Groover Clinic, Baptist Medical Center, St. Vincent Medical Center"
  ],
  "Boston Children's Hospital": [
    "Boston Children's Hospital",
    "Boston Children's Hospital II Harvard Medical School",
    "Boston Childrens Hospital",
    "Boston Children’s Hospital",
    "Boston Children’s Hospital//Boston Medical Center",
    "Boston Children’s Hospital/Beth Israel Deaconess Medical Center/Harvard Medical School",
    "Boston Children’s Hospital/Boston Medical Center",
    "Boston Children’s Hospital/Boston Medical Center A",
    "Boston Children’s Hospital/Harvard Medical School",
    "Boston Children’s/ Brigham & Women’s Hospital, Harvard Medical School Affiliated Hospitals"
  ],
  "Boston City Hospital": [
    "Boston City Hospital"
  ],
  "Boston College": [
    "Boston College",
    "Boston College",
    "Boston college"
  ],
  "Boston Eye Group/Boston Laser": [
    "Boston Eye Group/Boston Laser"
  ],
  "Boston Medical Center": [
    "Boston Medical Center"
  ],
  "Boston Psychoanalytic Society and Institute": [
    "Boston Psychoanalytic Institute",
    "Boston Psychoanalytic Society & Institute",
    "Boston Psychoanalytic Society & Institute",
    "Boston Psychoanalytic Society Institute",
    "Boston Psychoanalytic Society and Institute",
    "Boston Psychoanalytic Society and Institute",
    "Boston Psychoanalytic Society and Institute, Inc."
  ],
  "Boston State Hospital": [
    "Boston State Hospital",
    "Boston State Hospital/Dorchester Mental Health Center"
  ],
  "Boston University": [
    "Boston U - Tuft University",
    "Boston Univ Med Ctr",
    "Boston Univeristy",
    "Boston University",
    "Boston University",
    "Boston University College of Arts and Sciences",
    "Boston University College of Engineering",
    "Boston University College of Liberal Arts",
    "Boston University Graduate School of Management",
    "Boston University Medical Center",
    "Boston University Medical Center",
    "Boston University School of Medicine",
    "Boston University School of Medicine",
    "Boston University School of Public Health",
    "Boston University School of Public Health",
    "Boston University [Seven-Year BA/MD Program]",
    "Boston University – College of Arts & Sciences",
    "Boston University, College of Liberal Arts, Combined BA/MD Program",
    "Boston Univesity",
    "boston university"
  ],
  "Boston VA Healthcare System": [
    "Boston VA Healthcare System (Brockton-West Roxbury)",
    "Boston VA Healthcare System (Brockton-West Roxbury)/Harvard Medical School"
  ],
  "Botsford General Hospital": [
    "Bots ford General Hospital",
    "Botsford General Hospital"
  ],
  "Bowdoin College": [
    "Bowdoin College"
  ],
  "Bowling Green State University": [
    "Bowling Green State University"
  ],
  "Brackenridge Hospital": [
    "Brackenridge Hospital"
  ],
  "Bradford Regional Medical Center": [
    "Bradford Regional Medical Center"
  ],
  "Bradley University": [
    "Bradley University"
  ],
  "Braithwaite Memorial Hospital": [
    "Braithwaite Memorial Hospital"
  ],
  "Brandeis University": [
    "Brandeis",
    "Brandeis U",
    "Brandeis University",
    "Brandeis University",
    "Brandeis University - Heller School for Social Policy and Management",
    "Brandeis University- Heller School",
    "Brandeis-Heller School for Social Policy & Management"
  ],
  "Brasenose College, Oxford University": [
    "Brasenose College, Oxford University"
  ],
  "Brattleboro Retreat": [
    "Brattleboro Retreat"
  ],
  "Brebeuf College School": [
    "Brebeuf College School"
  ],
  "Brenau University": [
    "Brenau University"
  ],
  "Brentwood Hospital - Osteopathic": [
    "Brentwood Hospital - Osteopathic"
  ],
  "Brian Allgood Army Community Hospital(BAACH)/121st Combat Support Hospital (CSH), Republic of Korea (ROK)": [
    "Brian Allgood Army Community Hospital(BAACH)/121st Combat Support Hospital (CSH), Republic of Korea (ROK)"
  ],
  "Bridgeport Hospital": [
    "Bridgeport Hospital",
    "Bridgeport Hospital in Affiliation with Yale University School of Medicine",
    "Bridgeport Hospital/Yale University"
  ],
  "Bridgewater State Hospital": [
    "Bridgewater State Hospital"
  ],
  "Brigham and Women's Hospital": [
    "Brigham & Women's Hospital",
    "Brigham & Womens Hosp",
    "Brigham and Women's Faulkner Hospital",
    "Brigham and Women's Hospital",
    "Brigham and Women's Hospital",
    "Brigham and Women's Hospital A",
    "Brigham and Women's Hospital/Boston Children's Hospital",
    "Brigham and Women's Hospital/Children's Hospital",
    "Brigham and Women's Hospital/Children's Hospital/Harvard Medical School",
    "Brigham and Women's Hospital/Harvard Medical School",
    "Brigham and Women's Hospital/Massachusetts General Hospital",
    "Brigham and Women's Hospital/Massachusetts General Hospital/Dana-Farber Cancer Institute",
    "Brigham and Women's Hospital/Massachusetts General Hospital/Harvard Medical School",
    "Brigham and Women's Hospital/West Roxbury Veterans Affairs Medical Center",
    "Brigham and Womens Hospital",
    "Brigham and Women’s Hospital",
    "Brigham and Women’s/ Harvard Medical School"
  ],
  "Brighton Medical Center": [
    "Brighton Medical Center"
  ],
  "Bristol Health": [
    "Bristol Health"
  ],
  "British American Drama Academy": [
    "British American Drama Academy"
  ],
  "British Heart Failure Society , Royal Brompton and Harefield Hospital": [
    "British Heart Failure Society , Royal Brompton and Harefield Hospital"
  ],
  "British Psychoanalytical Society": [
    "British Psychoanalytical Society"
  ],
  "Broad Institute of MIT and Harvard": [
    "Broad Institute of MIT and Harvard"
  ],
  "Broadlawns Medical Center": [
    "Broadlawns Medical Center"
  ],
  "Brock University": [
    "Brock University"
  ],
  "Brody School of Medicine": [
    "Brody School of Medicine",
    "Brody School of Medicine at East Carolina University"
  ],
  "Bronx Care hospital center": [
    "Bronx Care hospital center"
  ],
  "Bronx Community College": [
    "Bronx Community College"
  ],
  "Bronx High School of Science": [
    "Bronx High School of Science"
  ],
  "Bronx Lebanon Hospital": [
    "Bronx Lebanon Hospital",
    "Bronx Lenanon Hospital",
    "Bronx-Lebanon Hospital"
  ],
  "Bronx Psychiatric Center": [
    "Bronx Psychiatric Center"
  ],
  "BronxCare Health System": [
    "BronxCare Health System"
  ],
  "Brookdale University Hospital and Medical Center": [
    "Brookdale Hosp Med Ctr",
    "Brookdale Hospital Medical Center",
    "Brookdale University Hospital and Medical Center"
  ],
  "Brooke Army Medical Center": [
    "Brooke Army Medical Center",
    "Brooke Army Medical Center - Fort Sam Houston, TX"
  ],
  "Brookline Community Health Center": [
    "Brookline Community Health Center"
  ],
  "Brooklyn": [
    "Brooklyn"
  ],
  "Brooklyn College": [
    "Brooklyn College",
    "Brooklyn College (CUNY)",
    "Brooklyn College 7yr. BA/MD program",
    "Brooklyn College, City University of New York",
    "Brooklyn College, City University of New York (CUNY)"
  ],
  "Brooklyn College of Pharmacy": [
    "Brooklyn College of Pharmacy"
  ],
  "Brooklyn Hospital Center": [
    "Brooklyn Eye Ear Hospital",
    "Brooklyn Hospital",
    "Brooklyn Hospital Center"
  ],
  "Brooklyn Jewish Hospital and St. John's Episcopal Medical Center ('Interfaith Medical Center')": [
    "Brooklyn Jewish Hospital and St. John's Episcopal Medical Center ('Interfaith Medical Center')"
  ],
  "Brooklyn Law School": [
    "Brooklyn Law School"
  ],
  "Brooklyn Womens Hospital": [
    "Brooklyn Womens Hospital"
  ],
  "Brooklyn-Cumberland Medical Center": [
    "Brooklyn-Cumberland Medical Center"
  ],
  "Brooks AFB": [
    "Brooks AFB",
    "Brooks Air Force Base"
  ],
  "Brookwood Baptist Health": [
    "Brookwood Baptist Health"
  ],
  "Brophy College Preparatory": [
    "Brophy College Preparatory"
  ],
  "Broward Health": [
    "Broward Health",
    "Broward Health Medical Center"
  ],
  "Brunswick Community College": [
    "Brunswick Community College"
  ],
  "Brussels Belgium Universite libre": [
    "Brussels Belgium Universite libre"
  ],
  "Bryan Medical Center": [
    "Bryan Medical Center"
  ],
  "Bryant University": [
    "Bryant University"
  ],
  "Bryn Mawr College": [
    "Bryn Mawr",
    "Bryn Mawr College",
    "Bryn Mawr Graduate School of Social Work & Social Research"
  ],
  "Bryn Mawr Hospital": [
    "Bryn Mawr Hospital"
  ],
  "Bucknell University": [
    "Bucknell University",
    "Bucknell University"
  ],
  "Buffalo General Hospital": [
    "Buffalo General Hospital"
  ],
  "Buffalo Psychiatric Center": [
    "Buffalo Psychiatric Center"
  ],
  "Buffalo University": [
    "Buffalo University"
  ],
  "Bukovinian State Medical University": [
    "Bukovinian Med Univ",
    "Bukovinian State Medical University"
  ],
  "Burdenko Neurosurgery Institute": [
    "Burdenko Neurosurgery Institute"
  ],
  "Bureau for Health Services": [
    "Bureau for Health Services"
  ],
  "Burnett Honors College, University of Central Florida": [
    "Burnett Honors College, University of Central Florida"
  ],
  "Burrell College of Osteopathic Medicine": [
    "Burrell College of Osteopathic Medicine"
  ],
  "Burroughs Wellcome Company": [
    "Burroughs Wellcome Company"
  ],
  "Busch Eye Institute - St. Louis University": [
    "Busch Eye Institute - St. Louis University"
  ],
  "Bush School of Government and Public Service": [
    "Bush School of Government and Public Service"
  ],
  "Butler Hospital": [
    "Butler Hospital",
    "Butler Hospital/Brown University"
  ],
  "Butler University": [
    "Butler University"
  ],
  "Butterworth Hospital": [
    "Butterworth Hospital"
  ]
}


In [231]:
total = 0
for entry in list(my_dict.values()):
    total += len(entry)

In [232]:
total 

455

In [147]:
len(list(my_dict.keys()))

230

In [180]:
my_dict['Baylor College of Medicine'] = my_dict.pop('BAYLOR')
my_dict['Beth Israel Deaconess Medical Center'] = my_dict.pop('B I Deaconess Med Ctr')
my_dict

{'B J Medical College': ['B J Medical College Gujarat University.',
  'B J Medical college, Ahmedabad , India',
  'B. J. Medical College Ahmedabad',
  'B.J. Medical College - University of Pune'],
 'B Palmer Eye Institute': ['B Palmer Eye Institute - A B Leach Eye',
  'Bascom Palmer Eye Institute',
  'Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami',
  'Bascom Palmer Eye Institute, University of Miami'],
 "B.J. Wadia Children's Hospital": ["B.J. Wadia Children's Hospital"],
 'B.P. Koirala Institute of Health Sciences': ['B.P. Koirala Institute of Health Sciences'],
 'BAMC': ['BAMC'],
 'BENEDICTINE UNIV': ['BENEDICTINE UNIV',
  'Benedictine College',
  'Benedictine Hospital',
  'Benedictine University'],
 'BLDE University - Medical College': ['BLDE University - Medical College'],
 'BMU Karachi': ['BMU Karachi'],
 'BRIGHAM YOUNG UNIVERSITY': ['BRIGHAM YOUNG UNIVERSITY',
  'BYU',
  'Brigham Young Univerdity',
  'Brigham Young University',
  'Brigham Young 

In [245]:
b_list_of_lists = []
b_list = []
for u,v in my_dict.items():
    my_list = [u]
    for item in v:
        my_list += [item]
    b_list_of_lists.append(my_list)
b_list_of_lists

b_list = []
for u,v in my_dict.items():
    b_list.append(u)
    for item in v:
        b_list.append(item)
b_list

['B J Medical College',
 'B J Medical College Gujarat University.',
 'B J Medical college, Ahmedabad , India',
 'B. J. Medical College Ahmedabad',
 'B.J. Medical College - University of Pune',
 'B Palmer Eye Institute',
 'B Palmer Eye Institute - A B Leach Eye',
 'Bascom Palmer Eye Institute',
 'Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami',
 'Bascom Palmer Eye Institute, University of Miami',
 "B.J. Wadia Children's Hospital",
 "B.J. Wadia Children's Hospital",
 'B.P. Koirala Institute of Health Sciences',
 'B.P. Koirala Institute of Health Sciences',
 'BAMC',
 'BAMC',
 'BENEDICTINE UNIV',
 'BENEDICTINE UNIV',
 'Benedictine College',
 'Benedictine Hospital',
 'Benedictine University',
 'BLDE University - Medical College',
 'BLDE University - Medical College',
 'BMU Karachi',
 'BMU Karachi',
 'BRIGHAM YOUNG UNIVERSITY',
 'BRIGHAM YOUNG UNIVERSITY',
 'BYU',
 'Brigham Young Univerdity',
 'Brigham Young University',
 'Brigham Young University - Idaho',


In [247]:
b_list

['B J Medical College',
 'B J Medical College Gujarat University.',
 'B J Medical college, Ahmedabad , India',
 'B. J. Medical College Ahmedabad',
 'B.J. Medical College - University of Pune',
 'B Palmer Eye Institute',
 'B Palmer Eye Institute - A B Leach Eye',
 'Bascom Palmer Eye Institute',
 'Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami',
 'Bascom Palmer Eye Institute, University of Miami',
 "B.J. Wadia Children's Hospital",
 "B.J. Wadia Children's Hospital",
 'B.P. Koirala Institute of Health Sciences',
 'B.P. Koirala Institute of Health Sciences',
 'BAMC',
 'BAMC',
 'BENEDICTINE UNIV',
 'BENEDICTINE UNIV',
 'Benedictine College',
 'Benedictine Hospital',
 'Benedictine University',
 'BLDE University - Medical College',
 'BLDE University - Medical College',
 'BMU Karachi',
 'BMU Karachi',
 'BRIGHAM YOUNG UNIVERSITY',
 'BRIGHAM YOUNG UNIVERSITY',
 'BYU',
 'Brigham Young Univerdity',
 'Brigham Young University',
 'Brigham Young University - Idaho',


In [198]:
my_dict_lower = {
    k.lower(): [v.lower() for v in val] for k, val in my_dict.items()
}
my_dict_lower

{'b j medical college': ['b j medical college gujarat university.',
  'b j medical college, ahmedabad , india',
  'b. j. medical college ahmedabad',
  'b.j. medical college - university of pune'],
 'b palmer eye institute': ['b palmer eye institute - a b leach eye',
  'bascom palmer eye institute',
  'bascom palmer eye institute - anne bates leach eye hospital, university of miami',
  'bascom palmer eye institute, university of miami'],
 "b.j. wadia children's hospital": ["b.j. wadia children's hospital"],
 'b.p. koirala institute of health sciences': ['b.p. koirala institute of health sciences'],
 'bamc': ['bamc'],
 'benedictine univ': ['benedictine univ',
  'benedictine college',
  'benedictine hospital',
  'benedictine university'],
 'blde university - medical college': ['blde university - medical college'],
 'bmu karachi': ['bmu karachi'],
 'brigham young university': ['brigham young university',
  'byu',
  'brigham young univerdity',
  'brigham young university',
  'brigham young 

In [224]:
canonical_names = list(my_dict_lower.keys())
process.extractOne('baylor uni', canonical_names, scorer=fuzz.partial_token_sort_ratio)

('baylor college of medicine', 82.35294117647058, 228)

In [226]:
canonical_names

['b j medical college',
 'b palmer eye institute',
 "b.j. wadia children's hospital",
 'b.p. koirala institute of health sciences',
 'bamc',
 'benedictine univ',
 'blde university - medical college',
 'bmu karachi',
 'brigham young university',
 'brown university',
 'baba raghav das medical college',
 'babson college',
 'baghdad university medical college',
 'bahcesehir university',
 'baker college',
 'baldwin wallace university',
 'ball state university',
 'balliol college',
 'baltimore city community college',
 'baltimore city hospital',
 'baltimore county fire-rescue academy',
 'baltimore medical college',
 'baltimore university',
 'baltimore washington institute for psychoanalysis',
 'banaras hindu university iom',
 'banff school of management',
 'bangabandhu shiekh mujib medical university',
 'bangalore medical college',
 'bangalore university',
 'bangladesh college of physicians and surgeons',
 'bangladesh medical college',
 'bank street college of education',
 'bankstown-lidcomb

In [228]:
q = {'d':[1,2,3], 'e':[4,5,6]}
q['d'].append(7)
q

{'d': [1, 2, 3, 7], 'e': [4, 5, 6]}

In [ ]:
inst_list = 
for i in m:
    if i not in b_list:
        print(i)

Baldwin Wallace University 
Banner University of Arizona 
Baptist Medical Center 
Barnard College of Columbia University 
Barnard College, Columbia U. 
Barnard College, Columbia University 
Baroda medical college 
Barrow Neurological Institute 
Barry University 
Baruch College/  CUNY
Baruch college/ Mt Sinai School of Medicine 
Bascom Palmer Eye Institute 
Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami 
Basildon and Thurrock  General Hospital
Baylor University 
Bearwood College 
Ben Gurion University 
Benemérita Universidad Autónoma de Puebla
Beth Israel Deaconess Medical Center 
Binghamton University 
Bloomberg School of Public Health, The Johns Hopkins University 
Boston College 
Boston Psychoanalytic Society &amp; Institute 
Boston Psychoanalytic Society and Institute 
Boston U - Tuft University 
Boston University 
Boston University Medical Center 
Boston University School of Medicine 
Boston University School of Public Health 
Boston university
Bra

In [241]:
b_list_of_lists = []
b_list = []
for u,v in my_dict.items():
    my_list = [u]
    for item in v:
        my_list += [item]
    b_list_of_lists.append(my_list)

b_list = []
for u,v in my_dict.items():
    b_list.append(u)
    for item in v:
        b_list.append(item)

my_dict_lower = {
    k.lower(): [v.lower() for v in val] for k, val in my_dict.items()
}

# Match each string to the most similar canonical name
for institution in institutions:
    match, score, *_ = process.extractOne(institution, canonical_names, scorer=fuzz.ratio)
    if score >= threshold:
        grouped_institutions[institution] = match
    else:
        # If no match is above the threshold, treat it as a new canonical name
        canonical_names.append(institution)
        grouped_institutions[institution] = institution

canonical_names = list(my_dict_lower.keys())
threshold = 80

for i in b_inst:
    if i not in b_list:
        name = i.strip()
        name = name.replace('\'', '')
        name = name.lower()
    if '/' in inst:
        name = name.split('/')[0]
    for word in [' hospital', ' university', ' medical', ' college', ' school']:
        name = name.replace(word, '') 
    
    match, score, *_ = process.extractOne(name, canonical_names, scorer=fuzz.token_ratio)
    if score >= threshold:
        my_dict_lower[match].append(i)
    else:
        my_dict_lower[i] = i

In [242]:
b_list_of_lists = []
b_list = []
for u,v in my_dict_lower.items():
    my_list = [u]
    for item in v:
        my_list += [item]
    b_list_of_lists.append(my_list)
b_list_of_lists

b_list = []
for u,v in my_dict_lower.items():
    b_list.append(u)
    for item in v:
        b_list.append(item)
b_list

['b j medical college',
 'b j medical college gujarat university.',
 'b j medical college, ahmedabad , india',
 'b. j. medical college ahmedabad',
 'b.j. medical college - university of pune',
 'b palmer eye institute',
 'b palmer eye institute - a b leach eye',
 'bascom palmer eye institute',
 'bascom palmer eye institute - anne bates leach eye hospital, university of miami',
 'bascom palmer eye institute, university of miami',
 'Bascom Palmer Eye Institute ',
 'Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami ',
 'Bascom Palmer Eye Institute, University of Miami',
 'Bashkir State Med Univ',
 "b.j. wadia children's hospital",
 "b.j. wadia children's hospital",
 'b.p. koirala institute of health sciences',
 'b.p. koirala institute of health sciences',
 'bamc',
 'bamc',
 'benedictine univ',
 'benedictine univ',
 'benedictine college',
 'benedictine hospital',
 'benedictine university',
 'blde university - medical college',
 'blde university - medical coll

In [243]:
for i in b_inst:
    if i not in b_list:
        print(i)

In [244]:
my_dict_lower

{'b j medical college': ['b j medical college gujarat university.',
  'b j medical college, ahmedabad , india',
  'b. j. medical college ahmedabad',
  'b.j. medical college - university of pune'],
 'b palmer eye institute': ['b palmer eye institute - a b leach eye',
  'bascom palmer eye institute',
  'bascom palmer eye institute - anne bates leach eye hospital, university of miami',
  'bascom palmer eye institute, university of miami',
  'Bascom Palmer Eye Institute ',
  'Bascom Palmer Eye Institute - Anne Bates Leach Eye Hospital, University of Miami ',
  'Bascom Palmer Eye Institute, University of Miami',
  'Bashkir State Med Univ'],
 "b.j. wadia children's hospital": ["b.j. wadia children's hospital"],
 'b.p. koirala institute of health sciences': ['b.p. koirala institute of health sciences'],
 'bamc': ['bamc'],
 'benedictine univ': ['benedictine univ',
  'benedictine college',
  'benedictine hospital',
  'benedictine university'],
 'blde university - medical college': ['blde univer

In [151]:
# Iterate through each element in b_inst
for i, institution in enumerate(b_inst):
    # Check if the institution is in any of the values of my_dict
    for key, values in my_dict.items():
        if institution in values:
            # Replace the element with the key corresponding to that value
            b_inst[i] = key
            break

# Print the updated b_inst
print(b_inst)

['B I Deaconess Med Ctr', 'B J Medical College', 'B J Medical College', 'B Palmer Eye Institute', 'B J Medical College', 'B J Medical College', "B.J. Wadia Children's Hospital", 'B.P. Koirala Institute of Health Sciences', 'BAMC', 'BAYLOR', 'BENEDICTINE UNIV', 'BLDE University - Medical College', 'BMU Karachi', 'BRIGHAM YOUNG UNIVERSITY', 'BROWN UNIVERSITY', 'BRIGHAM YOUNG UNIVERSITY', 'Baba Raghav Das Medical College', 'Babson College', 'Babson College', 'Babson College', 'Baghdad University Medical College', 'Bahcesehir University', 'Baker College', 'Baldwin Wallace University', 'Baldwin Wallace University ', 'Baldwin Wallace University', 'Ball State University', 'Balliol College', 'Baltimore City Community College', 'Baltimore City Hospital', 'Baltimore County Fire-Rescue Academy', 'Baltimore Medical College', 'Baltimore University', 'Baltimore Washington Institute for Psychoanalysis', 'Baltimore Washington Institute for Psychoanalysis', 'Baltimore Washington Institute for Psychoana

In [161]:
values

['BROWN UNIVERSITY, School for Professional Studies',
 'Brown',
 'Brown University',
 'Brown Univeristy',
 'Brown Univerity',
 'Brown University (Butler Hospital)',
 'Brown University Medical School',
 'Brown University School of Public Health',
 'Brown University, Associate Professor of Medicine',
 'Brown University, Class of 1982',
 'Brown University/ Women and Infants Hospital',
 'Brown University/Rhode Island Hospital-Lifespan',
 'Brown University/Women & Infants Hospital',
 'Brown/Rhode Island Hospital',
 'brown university']